Sources/Build/ManifestBuilder.swift

```diff
@@ -586,8 +586,9 @@ extension LLBuildManifestBuilder {
         for command in target.pluginInvocationResults.reduce([], { $0 + $1.buildCommands }) {
             // Create a shell command to invoke the executable. We include the path of the executable as a dependency.
             let execPath = AbsolutePath(command.configuration.executable, relativeTo: buildParameters.buildPath)
+            let uniquedName = ([command.configuration.displayName] + (command.inputFiles + command.outputFiles ).map{ $0.pathString }).joined(separator: "|")
```

Comment:
would it make sense to use something like UUID to make sure this is always unique, or do you want this to match if inputs + outputs are the same?

Reply:
The main thing is to make sure that's stable, so that running the same inputs generates the same llbuild manifest.  Otherwise we will have spurious rebuilds.  The combination of inputs and outputs seems sufficient to make it unique — if it's the same inputs and outputs then from llbuild's perspective there is nothing to distinguish when they run.

I suppose we could include the command line in the hash as well — I'm not really sure what happens if you have two commands with the same inputs and outputs — would you expect both commands to run?  Perhaps using the whole command line would be better.

Reply:
But UUIDs and things like that are usually the bane of build systems, since they introduce randomness, which generally defeats null builds, etc.

Reply:
In the general case, the question of whether a command that changes arguments from one run to the next is still the "same" command can be tricky.  It can for example affect how things are shown in IDEs, but it's unlikely to matter from a correctness standpoint here.  So even if, say, adding a "-v" flag from one run to the next doesn't change the identity of the command in a conceptual way, having it appear as a different command in the llbuild manifest might not cause any problems.  I'm therefore inclined to hash the whole command line.

Perhaps the plugin identity and the target to which it was applied should factor in here, although it should be noted that they are already baked into the output paths, since every combination of plugin and target is given its own subdirectory under the intermediates directory.

Reply:
Not entirely sure what the plain llbuild behavior is, but XCBuild will error if there are multiple commands producing the same output file.

Reply:
In this case the llbuild manifest logic was discarding the older commands.  In the case of a plugin that emits two commands with the same outputs, we should probably catch that earlier and emit an error.  I'll write up an enhancement request for that.